qiskit/synthesis/qft/qft_decompose_lnn.py

```diff
@@ -0,0 +1,68 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Circuit synthesis for a QFT circuit.
+"""
+
+import numpy as np
+from qiskit.circuit import QuantumCircuit
+from qiskit.synthesis.linear_phase.cz_depth_lnn import _append_cx_stage1, _append_cx_stage2
+
+
+def synth_qft_line(num_qubits, do_swaps=True, approximation_degree=0):
+    """Synthesis of a QFT circuit for a linear nearest neighbor connectivity.
+    Based on Fig 2.b in Fowler et al. [1].
+
+    Note that this method *reverts* the order of qubits in the circuit,
+    compared to the original QFT code.
+    Hence, the default value of do_swaps parameter is True,
+    since it produces a circuit with fewer CX gates.
+
+    Args:
+        num_qubits (int): The number of qubits on which the QFT acts.
+        approximation_degree (int): The degree of approximation (0 for no approximation).
+        do_swaps (bool): Whether to include the final swaps in the QFT.
+
+    Return:
+        QuantumCircuit: a circuit implementation of the QFT circuit.
+
+    Reference:
+        1. A. G. Fowler, S. J. Devitt, and L. C. L. Hollenberg,
+           *Implementation of Shor's algorithm on a linear nearest neighbour qubit array*,
+           Quantum Info. Comput. 4, 4 (July 2004), 237–251.
+           `arXiv:quant-ph/0402196 [quant-ph] <https://arxiv.org/abs/quant-ph/0402196>`_
+    """
+
+    qc = QuantumCircuit(num_qubits)
+
+    for i in range(num_qubits):
+        num_entanglements = max(0, max(0, approximation_degree - (num_qubits - i - 1)))
+
+        qc.h(num_qubits - 1)
+        for j in range(i, num_qubits - num_entanglements - 1):
+            qc.p(np.pi / 2 ** (j - i + 2), num_qubits - j + i - 1)
+            qc.cx(num_qubits - j + i - 1, num_qubits - j + i - 2)
+            qc.p(-np.pi / 2 ** (j - i + 2), num_qubits - j + i - 2)
+            qc.cx(num_qubits - j + i - 2, num_qubits - j + i - 1)
+            qc.cx(num_qubits - j + i - 1, num_qubits - j + i - 2)
+            qc.p(np.pi / 2 ** (j - i + 2), num_qubits - j + i - 1)
+
+    if not do_swaps:
+        # Add a reversal network for LNN connectivity in depth 2*n+2,
+        # based on Kutin at al., https://arxiv.org/abs/quant-ph/0701194, Section 5.
+        for _ in range((num_qubits + 1) // 2):
+            qc = _append_cx_stage1(qc, num_qubits)
+            qc = _append_cx_stage2(qc, num_qubits)
```

Comment:
We can leave as is, but it would be a bit nicer if we would not have to use a private method and could instead make it public

Reply:
I think that these are only small helper functions and should not be public.
Maybe we can add a function that does a reverse permutation for LNN in depth 2n? Then we can use it instead. 
Perhaps in a different PR?

Reply:
Sounds good, could you open an issue so we can keep track of this?